lib/mix/test/mix/release_test.exs

```diff
@@ -110,6 +110,18 @@ defmodule Mix.ReleaseTest do
       end
     end
 
+    test "uses the latest version of an app if there are multiple versions", context do
+      in_tmp(context.test, fn ->
+        File.cp_r!(:code.root_dir(), ".", fn _, _ -> false end)
+        File.mkdir_p!("lib/compiler-1.0")
+        erts_source = Path.join(File.cwd!(), "erts-#{@erts_version}")
+
+        release = from_config!(nil, config(releases: [demo: [include_erts: erts_source]]), [])
+
+        assert release.applications.compiler[:vsn] != "1.0"
```

Comment:
Let's also assert `assert release.applications.compiler[:path]` points to the current working dir just in case. :)